betty/console/__init__.py

```diff
@@ -0,0 +1,500 @@
+"""
+The Betty console.
+"""
+
+import argparse
+import logging
+import sys
+from asyncio import run
+from collections.abc import Callable, Iterable, Sequence
+from contextlib import AsyncExitStack, suppress
+from enum import IntEnum
+from pathlib import Path
+from traceback import format_exception
+from typing import Any, Never, TextIO, TypeVar, cast, final, overload
+
+import rich_argparse
+from rich.console import Console
+from rich.prompt import Confirm, Prompt
+from typing_extensions import override
+
+from betty import about
+from betty.app import App
+from betty.assertion import Assertion, assert_path
+from betty.assertion.error import AssertionFailed
+from betty.config import assert_configuration_file
+from betty.console import commands
+from betty.console.commands import Command, CommandFunction
+from betty.error import FileNotFound, UserFacingError
+from betty.locale.localizable import Localizable, _
+from betty.locale.localizer import DEFAULT_LOCALIZER, Localizer
+from betty.logging import UserHandler
+from betty.project import Project
+from betty.project.config import ProjectConfiguration
+from betty.serde.format import FORMAT_REPOSITORY
+from betty.typing import Void, internal
+from betty.user import User, Verbosity
+
+_T = TypeVar("_T")
+
+
+class ExitCode(IntEnum):
+    """
+    The exit codes used by the console.
+
+    The keys are stable. The values are internal and may change without notice.
```

Comment:
The values are NOT internal. In fact they are very, very much part of the public API.